cmd/posix.go

```diff
@@ -911,9 +911,20 @@ func (s *posix) ListDir(volume, dirPath string, count int, leafFile string) (ent
 		return nil, err
 	}
 
+	if mustPrefix != "" {
+		entries = filterMatchingPrefix(entries, mustPrefix)
```

Comment:
We cannot prematurely do this because of how the flat namespace prefix matching would work with or without '/' as this may match entries which are not correct.

Reply:
The only thing that can happen to entry is that it can have a slash trimmed before it is returned to `listDirFactory`. Note that `listDirFactory` still does the filter.

So if entry is `prefix/file/` at this point, the a "mustPrefix" of "prefix/file" would still accept let it through and if we want prefix `prefix/file/` it will be filtered out in `listDirFactory`.

`listDirFactory` is the only function that sets `leafFile` (and `mustPrefix`).